File: holoken-core/src/main/java/com/srlee/dlx/DLXNode.java
package com.srlee.dlx;

class DLXNode extends LL2DNode {
	private final DLXColumn column;
	private final int rowIndex;
	
	DLXNode(final DLXColumn column, final int rowIndex) {
		this.rowIndex = rowIndex;
		this.column = column;
		column.up.down = this;
		up = column.up;
		down = column;
		column.up = this;
		column.incrementSize();
	}
	
	DLXColumn getColumn() {
		return column;
	}
	
	int getRow() {
		return rowIndex;
	}
}

File: holoken-core/src/main/kotlin/com/srlee/dlx/DLXNode.kt
package com.srlee.dlx

internal class DLXNode(val column: DLXColumn, val row: Int) : LL2DNode() {

    init {
        column.up!!.down = this
        up = column.up
        down = column
        column.up = this
        column.incrementSize()
    }
}
